pkg/agent/protocols/manager.go

```diff
@@ -0,0 +1,15 @@
+package protocols
+
+type ProtocolManager interface {
+	Download(modelDir string, storageUri string) error
```

Comment:
Typically a download or copy or move API would be <src, dst>.
I'd probably have your storage implementation be model agnostic w/ it's names
e.g. Download(src, dest)

Reply:
I would have prefered this as well, I can do storageUri, and build the `dest` for you. But the `dest` is a combo of modelDir and modelName